### ISSUE TITLE
[Request] Battery indicator only mode

### ISSUE THREAD

**Issue by itssapir** [Ryochan7/DS4Windows#455]
Hi,
usually i don't use DS4Windows since my controllers are detected fine without it,
but this is the only software i could find which lets you see the battery percentage of your ds4 controllers in windows.
If it is not too much of a hassle, is it possible to make this a stand alone feature?
as in, be able to open ds4windows, with it not reconfiguring anything, but still have it show the battery of the controllers?

**Comment by IceLancerSR** [comment]
Man, if i had money to offer i would have ! I already posted request on reddit before about this. Just a tool with features , without changing gamepad into xbox .. it would be.. i am out of words it would be simply awesome!

**Comment by Ryochan7** [comment]
This is already possible with USB and Sony Wireless Adapter connections. You can just have a default profile with the "Use DInput Only" option enabled which will unplug a virtual Xbox 360 controller associated with that slot and nullify any non-KB+M mappings; you would also have to make sure to use shared mode instead of exclusive mode. Bluetooth is the main obstacle since the routines used on the HID device change the input mode for the controller and make the old DirectInput controller inaccessible to any process that does not already have an open handle on it. There is no way to get around that problem without getting rid of a lot of features including being able to change the LED light which is commonly used to gauge battery life.

**Comment by IceLancerSR** [comment]
meh..it keeps saying in log, "using shared mode" and gamepad is only working in ds4windows app, nowhere else.. :(

**Comment by Ryochan7** [comment]
That behavior is likely due to one of two things. The first is that using a Bluetooth connection with DS4Windows will pretty much disable the original exposed controller in Windows although it should still be visible when using shared mode. The second possibility is that maybe you have HidGuardian installed on your system which could hide the originally controller from the system but DS4Windows would still be able to use it. That would only apply if you installed HidGuardian from some other program (InputMapper?) and you run DS4Windows as admin.

**Comment by IceLancerSR** [comment]
I did install hidguardian 4.1.0 (last version) from owner not from input mapper

**Comment by Ryochan7** [comment]
I have not tried HidGuardian 4.x series. I don't use HidGuardian on a regular basis but I still have version 1.14.0.0 installed on my system. It seems that every later release was intentionally marked beta and not ready for use in production. It does not help that docs are not readily available regarding recent changes to how HidGuardian works. If it weren't for some use cases and problems with the older method, I would probably discourage people from using HidGuardian at this point.

**Comment by Ryochan7** [comment]
Even going on the ViGEm Discord server didn't help much as far as obtaining documentation. Unlike with HG 1.x series, it looks like the 4.x series will depend on a separate service running on your system that user-mode applications will interact with in order to gain access to devices. As it is now, HG 4.x does not have a "stable" signed release that I can see. Time will tell whether this will negatively impact DS4Windows or not or if I will bother moving to the 4.x series of the driver.

It looks like the current methods being used in DS4Windows to interact with HidGuardian can be removed in the future because they will be useless with the 4.x series driver.

**Comment by Ryochan7** [comment]
Update. I tried out HidGuardian 4.1.126.0 and the latest HidVigil source. I could not figure out how to get DS4Windows whitelisted so it could access blocked devices so an exemption registry entry was added instead. However, it did not actually matter. HidGuardian 4.x runs terribly in general on my system. My whole PC becomes less responsive and normal KB+M operation is more laggy while the driver is enabled. Disabling the driver device makes my system run normally again.
